test/e2e/framework/prometheus/prometheus.go

```diff
@@ -58,6 +58,31 @@ func CheckMetric(promAddress, metricName string, validMetric map[string]string)
 	return nil
 }
 
+func GetMetricGuageValueFromBuffer(prometheusMetricData []byte, metricName string, expectedLabels map[string]string) (float64, error) {
+	metrics, err := getAllPrometheusMetricsFromBuffer(prometheusMetricData)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse prometheus metrics: %w", err)
+	}
+
+	for _, metric := range metrics {
```

Comment:
suggest: Try not to shadow variables. Can we rename one of the 'metric' variables?

Also just checking here - there's no good way to remove any of these for loops for efficiency, right?